File: cheaters/PacketLife/nitro-python-1.0/apidoc/docs/samples/ns-java-sample/rba.java
package NSConfig;
import java.io.*;
import java.util.*;
import java.net.*;
import java.beans.*;
import java.net.*;
import NSConfig.*;
import org.apache.axis.types.*;
    
public class rba  {
	public static void main(String[] args) {
	   String  soap_server;
	   String username;
	   String password ;
    
	   NSConfigBindingStub client ;
	   SimpleResult  result;
	   UnsignedInt NULL_UINT=null;
	   if (args.length < 3) {
	      System.out.println("Usage : java setConfig.class <NP> <username> <password>");
	      return;
	   }
	   soap_server = args[0];
	   username    = args[1];
	   password    = args[2];
	   try {
            			// Initialized the client stub
				soap_server = "http://"+soap_server+"/soap";
				System.out.println("Connecting to  "+soap_server + " ........");
	  
				client  = new NSConfigBindingStub(new URL(soap_server),null);
			    client.setMaintainSession(true);
				((org.apache.axis.client.Stub)client)._setProperty(org.apache.axis.AxisEngine.PROP_DOMULTIREFS, Boolean.FALSE);
				//Login
				result = client.login(username, password);
				System.out.println("Login : "+result.getMessage());
          
				result = client.addsystemuser("user1","passwd1");
				System.out.println("Addsystemuser	user1 : " + result.getMessage());

				result = client.addsystemuser("user2","passwd2");
				System.out.println("Addsystemuser	user2 : " + result.getMessage());

				result = client.bindsystemuser_policy("user1","read-only",new UnsignedInt(1));
				System.out.println("Bindsystemuser_policy	user1 : " + result.getMessage());

				result = client.bindsystemuser_policy("user1","operator",new UnsignedInt(2));
				System.out.println("Bindsystemuser_policy	user1 : " + result.getMessage());

				result = client.addsystemgroup("group1");
				System.out.println("Addsystemgroup	group1 : " + result.getMessage());

				result = client.bindsystemgroup_user("group1","user2");
				System.out.println("Bindsystemgroup_user	group1 : " + result.getMessage());

				result = client.bindsystemgroup_policy("group1","read-only",new UnsignedInt(1));
				System.out.println("Bindsystemgroup_policy	group1 : " + result.getMessage());

				result = client.addsystemcmdpolicy("add_pol1",(AllowdenyEnum)AllowdenyEnum.ALLOW,"^add.*");
				System.out.println("Addsystemcmdpolicy	add_pol1 : " + result.getMessage());

				result = client.bindsystemuser_policy("user1","add_pol1",new UnsignedInt(3));
				System.out.println("Bindsystemuser_policy	user1 : " + result.getMessage());

				GetsystemuserResult  Getsystemuserresult1 = client.getsystemuser(null);
				System.out.println("Getsystemuser	null : " + Getsystemuserresult1.getMessage());

				if ( (Getsystemuserresult1.getList() != null) && (Getsystemuserresult1.getList().length > 0)) {
					for(int i =0 ; i < Getsystemuserresult1.getList().length;i++) { 
						Systemuser obj;
						obj = (Systemuser) Getsystemuserresult1.getList()[i];
						System.out.println("\tusername\t\t =  "+obj.getUsername()+"\n"+"\tgroupname\t\t =  "+obj.getGroupname()+"\n"+"\tpolicyname\t\t =  "+obj.getPolicyname()+"\n"+"\tpriority\t\t =  "+obj.getPriority()+"\n");
					}
				}

				GetsystemgroupResult  Getsystemgroupresult2 = client.getsystemgroup(null);
				System.out.println("Getsystemgroup	null : " + Getsystemgroupresult2.getMessage());

				if ( (Getsystemgroupresult2.getList() != null) && (Getsystemgroupresult2.getList().length > 0)) {
					for(int i =0 ; i < Getsystemgroupresult2.getList().length;i++) { 
						Systemgroup obj;
						obj = (Systemgroup) Getsystemgroupresult2.getList()[i];
						System.out.println("\tgroupname\t\t =  "+obj.getGroupname()+"\n"+"\tusername\t\t =  "+obj.getUsername()+"\n"+"\tpolicyname\t\t =  "+obj.getPolicyname()+"\n"+"\tpriority\t\t =  "+obj.getPriority()+"\n");
					}
				}

				result = client.logout();
				System.out.println("logout : " + result.getMessage());
	   }
	   catch (Exception ex) {
	       ex.printStackTrace();
	   }
	}

}

File: cheaters/PacketLife/nitro-python-1.0/apidoc/docs/samples/ns-csharp-sample/rba.cs
using System;
using System.Runtime.Remoting;
using System.Xml.Serialization;
using System.Web.Services;
    
	namespace NSConfig {
		public class ClientService : NSConfigService

                {
                        private string cookie = null;
                        /* override the getWebRequest to send cookie */
                        protected override System.Net.WebRequest  GetWebRequest(Uri uri)
                        {
                                System.Net.HttpWebRequest req = (System.Net.HttpWebRequest) base.GetWebRequest(uri);
                                if (cookie != null)
                                {
                                        req.Headers.Add("Set-Cookie", cookie);
                                }
                                return req;
                        }
                        /* override the getWebResponse to get the cookie */

                        protected override System.Net.WebResponse GetWebResponse(System.Net.WebRequest req)
                        {
                               System.Net.HttpWebResponse rep = (System.Net.HttpWebResponse) base.GetWebResponse(req);
                               if (rep.Headers["Set-Cookie"] != null)
                                {
                                       cookie = rep.Headers["Set-Cookie"];
                                }
                                return rep;
			}
                        public ClientService()
                        {
                                // Change the IP address pointing to netscaler.
                                this.Url = "http://10.102.4.111:18000";
                        }
                        public ClientService(string servername)
                        {
                                this.Url = "http://"+servername+"/soap";;
                        }
                }
	
	class rba 
	{
 		static ClientService client=null;
 
		[STAThread]
		static void Main(string[] args)
		{
			if ( args.Length < 3)
			{
				Console.WriteLine("Usage: getConfig <NS IP> username password");
				return;
			}
			string serverip = args[0];
			string username = args[1];
			string password = args[2];
			try {
          			Console.WriteLine("\nConnecting to server "+serverip+" ............\n");
				client = new ClientService(serverip);

				client.CookieContainer = new System.Net.CookieContainer();
				simpleResult result = client.login(username,password) ;
				Console.WriteLine("login : "+result.message);
       
				result = client.addsystemuser("user1","passwd1");
				Console.WriteLine("addsystemuser	user1 : " + result.message);

				result = client.addsystemuser("user2","passwd2");
				Console.WriteLine("addsystemuser	user2 : " + result.message);

				result = client.bindsystemuser_policy("user1","read-only",1);
				Console.WriteLine("bindsystemuser_policy	user1 : " + result.message);

				result = client.bindsystemuser_policy("user1","operator",2);
				Console.WriteLine("bindsystemuser_policy	user1 : " + result.message);

				result = client.addsystemgroup("group1");
				Console.WriteLine("addsystemgroup	group1 : " + result.message);

				result = client.bindsystemgroup_user("group1","user2");
				Console.WriteLine("bindsystemgroup_user	group1 : " + result.message);

				result = client.bindsystemgroup_policy("group1","read-only",1);
				Console.WriteLine("bindsystemgroup_policy	group1 : " + result.message);

				result = client.addsystemcmdpolicy("add_pol1",allowdenyEnum.ALLOW,"^add.*");
				Console.WriteLine("addsystemcmdpolicy	add_pol1 : " + result.message);

				result = client.bindsystemuser_policy("user1","add_pol1",3);
				Console.WriteLine("bindsystemuser_policy	user1 : " + result.message);

				getsystemuserResult  getsystemuserresult1 = client.getsystemuser(null);
				Console.WriteLine("getsystemuser	null : " + getsystemuserresult1.message);

				if ( (getsystemuserresult1.List != null) && (getsystemuserresult1.List.Length > 0)) {
					for(int i =0 ; i < getsystemuserresult1.List.Length;i++) { 
						systemuser obj;
						obj = (systemuser) getsystemuserresult1.List[i];
						Console.WriteLine( obj.username + "\t" + obj.groupname + "\t" + obj.policyname + "\t" + obj.priority + "\t" );
					}
				}

				getsystemgroupResult  getsystemgroupresult2 = client.getsystemgroup(null);
				Console.WriteLine("getsystemgroup	null : " + getsystemgroupresult2.message);

				if ( (getsystemgroupresult2.List != null) && (getsystemgroupresult2.List.Length > 0)) {
					for(int i =0 ; i < getsystemgroupresult2.List.Length;i++) { 
						systemgroup obj;
						obj = (systemgroup) getsystemgroupresult2.List[i];
						Console.WriteLine( obj.groupname + "\t" + obj.username + "\t" + obj.policyname + "\t" + obj.priority + "\t" );
					}
				}

				result = client.logout();
				Console.WriteLine("logout : " + result.message); 
			}
			catch (Exception ex) {
			      Console.WriteLine( "Exception: "+ex.ToString());
			}
		}

	}
	}
